src/main/java/com/ecom/shopping/adminkart/service/ProductCategoryService.java

```diff
@@ -16,4 +19,24 @@ public List<ProductCategoryDto> getAllCategory() {
         return productCategoryMapper.findAllCategory();
     }
 
+    public ProductCategoryDto updateCategoryById(ProductCategoryDto productCategoryDto) {
+        Optional<ProductCategoryDto> existingUser = productCategoryMapper.getId(productCategoryDto.getId());
+        ProductCategoryDto updatedUser = null;
+        if (existingUser.isPresent()) {
+            updatedUser = existingUser.get();
+            updatedUser.setId(productCategoryDto.getId());
+            updatedUser.setName(productCategoryDto.getName());
+            updatedUser.setDescription(productCategoryDto.getDescription());
+            updatedUser.setModifiedDate(getDateTime());
+            productCategoryMapper.updateCategoryById(productCategoryDto.getId(), productCategoryDto.getName(), productCategoryDto.getDescription(), updatedUser.getModifiedDate());
```

Comment:
need to check if update count is greter than 1 then will return response , otherwise throw the exception